scipy/stats/mstats_basic.py

```diff
@@ -1282,34 +1300,35 @@ def ks_twosamp(data1, data2, alternative="two-sided"):
         Corresponding p-value.
 
     """
-    (data1, data2) = (ma.asarray(data1), ma.asarray(data2))
-    (n1, n2) = (data1.count(), data2.count())
-    n = (n1*n2/float(n1+n2))
-    mix = ma.concatenate((data1.compressed(), data2.compressed()))
-    mixsort = mix.argsort(kind='mergesort')
-    csum = np.where(mixsort < n1, 1./n1, -1./n2).cumsum()
-    # Check for ties
-    if len(np.unique(mix)) < (n1+n2):
-        csum = csum[np.r_[np.diff(mix[mixsort]).nonzero()[0],-1]]
-
-    alternative = str(alternative).lower()[0]
-    if alternative == 't':
-        d = ma.abs(csum).max()
-        prob = special.kolmogorov(np.sqrt(n)*d)
-    elif alternative == 'l':
-        d = -csum.min()
-        prob = np.exp(-2*n*d**2)
-    elif alternative == 'g':
-        d = csum.max()
-        prob = np.exp(-2*n*d**2)
-    else:
-        raise ValueError("Invalid value for the alternative hypothesis: "
-                         "should be in 'two-sided', 'less' or 'greater'")
+    # Ideally this would be accomplished by
+    # ks_2samp = scipy.stats.stats.ks_2samp
+    # but the circular dependencies between mstats_basic and stats prevent that.
+    alternative = {'t': 'two-sided', 'g': 'greater', 'l': 'less'}.get(
+       alternative.lower()[0], alternative)
+    return scipy.stats.stats.ks_2samp(x, y, alternative=alternative, mode=mode)
+
 
-    return (d, prob)
+ks_twosamp = ks_2samp
 
 
-ks_2samp = ks_twosamp
+def kstest(data1, data2, args=(), alternative='two-sided', mode='auto'):
```

Comment:
in ks_2samp / ks_twosamp, data1 and data2 are replaced by x and y whereas kstest still uses data1 and data2. is that intendend?

Reply:
the change in ks_twosamp could even be a problem with backwards compatability?

Reply:
Good spot.  The arguments could be provided by name to `ks_2samp`, even though they are required args.  In `kstest` it is not an issue as `kstest` was not exposed before now.